test/e2e/tests/connections/multiple-standard-connections.spec.ts

```diff
@@ -0,0 +1,315 @@
+          .withPermissionControllerConnectedToTestDapp({
+            account: EVM_ACCOUNT_TWO,
+          })
+          .build(),
+        title: this.test?.fullTitle(),
+      },
+      async ({ driver }) => {
+        await loginWithoutBalanceValidation(driver);
+        const testDapp = new TestDapp(driver);
+        const connectAccountConfirmation = new ConnectAccountConfirmation(
+          driver,
+        );
+
+        await testDapp.openTestDappPage();
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.TestDApp);
+        await testDapp.check_pageIsLoaded();
+
+        await testDapp.check_connectedAccounts(EVM_ACCOUNT_TWO);
+
+        const requestPermissionsWithAccount1 =
+          getRequestPermissionsRequestObject({
+            caveats: [
+              {
+                type: 'restrictReturnedAccounts',
+                value: [EVM_ACCOUNT_ONE],
+              },
+            ],
+          });
+
+        await driver.executeScript(
+          `window.ethereum.request(${requestPermissionsWithAccount1})`,
+        );
+
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+
+        await connectAccountConfirmation.check_pageIsLoaded();
+
+        await connectAccountConfirmation.check_isAccountDisplayed(
+          EVM_ACCOUNT_ONE,
+        );
+        await connectAccountConfirmation.check_isAccountDisplayed(
+          EVM_ACCOUNT_TWO,
+        );
+
+        await connectAccountConfirmation.confirmConnect();
+
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.TestDApp);
+
+        const expectedConnectedAccounts = `${EVM_ACCOUNT_TWO.toLowerCase()},${EVM_ACCOUNT_ONE.toLowerCase()}`;
+        await testDapp.check_connectedAccounts(expectedConnectedAccounts);
+      },
+    );
+  });
+
+  it('should retain EVM permissions when connecting through the Solana Wallet Standard', async function () {
+    await withSolanaAccountSnap(
+      {
+        ...DEFAULT_SOLANA_TEST_DAPP_FIXTURE_OPTIONS,
+        title: this.test?.fullTitle(),
+        fixtureModifications: (builder) =>
+          builder
+            .withKeyringControllerAdditionalAccountVault()
+            .withPreferencesControllerAdditionalAccountIdentities()
+            .withAccountsControllerAdditionalAccountIdentities()
+            .withPermissionControllerConnectedToTestDappWithTwoAccounts(),
+      },
+      async (driver) => {
+        const testDapp = new TestDappSolana(driver);
+
+        await testDapp.openTestDappPage();
+        await testDapp.switchTo();
+
+        await connectSolanaTestDapp(driver, testDapp);
+
+        await driver.switchToWindowWithTitle(
+          WINDOW_TITLES.ExtensionInFullScreenView,
+        );
+
+        const sitePermissionPage = await getPermissionsPageForHost(
+          driver,
+          DAPP_HOST_ADDRESS,
+        );
+
+        await sitePermissionPage.check_connectedAccountsNumber(3);
+        await sitePermissionPage.check_connectedNetworksNumber(2);
```

Comment:
thoughts on checking explicitly which accounts and networks are connected?

Reply:
forgot about this bit, will add tomorrow morning